src/misc/debug.c

```diff
@@ -97,7 +97,15 @@ void debug_vprintf_trace(const char* func_name,
 	}
 
 	if (level <= debug_level) {
-
+#ifdef USE_LOG_BACKEND
+		char fmt_copy[1024] = "";
+		strcpy(fmt_copy, fmt);
+		// take care of the trailing newline often present...
+		if (fmt_copy[strlen(fmt)-1] == '\n') {
+			fmt_copy[strlen(fmt)-1] = '\0';
+		}
```

Comment:
This should then go into the new debug_printf_trace function and could directly call into the vendor wrappers. Or why do we need this output wrappers?

Reply:
Done.